django_celery_beat/models.py

```diff
@@ -138,6 +132,7 @@ class Meta:
         verbose_name = _('interval')
         verbose_name_plural = _('intervals')
         ordering = ['period', 'every']
+        unique_together = ('every', 'period')
```

Comment:
What if I want two different jobs to run on the same interval?  This would disallow that, that seems like an issue.

Reply:
I think there is a mixup between jobs and schedules here: You assign the same "every 4 days" schedule to two distinct jobs and they both run at the time. That works before and after this pull request.

The current `cls.objects.filter(**spec).delete()` is effectively a unique constraint, except that it might cascade destroy data and does not prevent creation in the first place.

I think what the code needs is one of:
- a) A uniqueness constraint for _all_ schedules (as done in the pull request)
- b) A uniqueness constraint for _none_ of the schedules and no dangerous  `cls.objects.filter(**spec).delete()` either

What do you think?

Reply:
> I think there is a mixup between jobs and schedules here: You assign the same "every 4 days" schedule to two distinct jobs and they both run at the time. That works before and after this pull request.
> 
> The current `cls.objects.filter(**spec).delete()` is effectively a unique constraint, except that it might cascade destroy data and does not prevent creation in the first place.
> 
> I think what the code needs is one of:
> 
>     * a) A uniqueness constraint for _all_ schedules (as done in the pull request)
> 
>     * b) A uniqueness constraint for _none_ of the schedules and no dangerous  `cls.objects.filter(**spec).delete()` either
> 
> 
> What do you think?

just curious if we add UniqueContraint Check here would that be any useful?

Reply:
Hi @auvipy I would need get back into the whole topic since it's been so long ago by now. I can do that work but I would some "guarantee" that we'll actually get somewhere with this in the next few weeks to not invest in a dead end. (In case you're a friend of voice meetings for stuff like that, we can definitely do a Jitsi or Google Meet on the topic if you like, but no pressure, just one more option.)

Reply:
Ok, I will release a new version tomorrow and do some more analysis to get back to you to reach a consensus